charts/observability-pipelines-worker/templates/deployment.yaml

```diff
@@ -0,0 +1,43 @@
+{{- if (eq .Values.role "Stateless-Aggregator") -}}
```

Comment:
@lukesteensen @binarylogic similar to the agent, if we wanted to force StatefulSets for durability reasons (though they can be created without persistent storage) we could drop this file and related templates.

Reply:
I'm not too sure about this one. AFAIK OP will support the normal range of disk-based Vector features, so I'm not sure if we want to differ here or not. Maybe @binarylogic will have thoughts.